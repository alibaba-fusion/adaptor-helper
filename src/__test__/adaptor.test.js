import { normalizeAdaptor, generateDemos, getDefaultProps, findProp, findOptionLabel } from '../adaptor';
import Types from '../types';

const getButtonProps = (props, state) => {
  return {
    type: props.level,
    loading: state === 'active',
    block: props.block,
    size: props.size,
    disabled: state === 'disabled',
    active: state === 'hover',
  };
};


const LEVELS_MAP = {
  normal: ['normal', 'primary', 'secondary', 'warning', 'light', 'dark'],
  text: ['normal', 'primary'],
  list: ['normal', 'primary', 'secondary'],
};

const SIZES_MAP = {
  normal: ['large', 'medium', 'small'],
  text: ['medium', 'small'],
  list: ['medium'],
};

const Button = {
  name: 'Button',
  shape: ['normal', 'text', 'list'], // default ['normal']
  editor(shape) {
    return {
      props: [{
        name: 'level',
        type: Types.enum,
        options: LEVELS_MAP[shape],
        default: 'normal',
      }, {
        name: 'size',
        type: Types.enum,
        options: SIZES_MAP[shape],
        default: 'medium',
      }, {
        name: 'block',
        type: Types.bool,
      }].filter(({ name }) => {
        return shape === 'normal' || name !== 'block';
      }),
      data: {
        icon: true,
        disable: true,
        active: true,
        hover: true,
        default: 'Hello',
      },
    };
  },
  content(shape) {
    const options = [
      {
        name: 'icon',
        options: ['none', 'left', 'right'],
        default: 'none',
      },
    ];

    if (shape !== 'text') {
      options.push({
        name: 'group',
        options: ['yes', 'no'],
        default: 'no',
      });
    }

    return {
      options,
      transform(props, { icon, group }) {
        const p = {
          ...props,
        };

        if (icon === 'left') {
          p.data = hasStateMark(p.data) ? `${p.data.substring(0, 1)}[arrow-left]${p.data.substring(1)}` : `[arrow-left]${p.data}`;
        } else if (icon === 'right') {
          p.data = `${p.data}[arrow-right]`;
        }

        if (group === 'yes') {
          p.data = [p.data, p.data].join('\n');
        }

        return p;
      },
    };
  },
  adaptor(props) {
    return null;
  },
  demoOptions(props) {
    let background = 'transparent';
    if (props.level === 'light') {
      background = '#EBECF0';
    } else if (props.level === 'dark') {
      background = '#000';
    }

    return {
      background,
    };
  },
};

describe('adaptor.js', () => {
  it('should be normalizeAdaptor', () => {
    const normalize = normalizeAdaptor(Button);

    expect(normalize).toMatchSnapshot();
    normalize.shape.forEach(({ value }) => {
      expect(normalize.editor(value)).toMatchSnapshot();
      expect(normalize.content(value)).toMatchSnapshot();
    });
  });

  it('should be getDefaultProps & finds', () => {
    const normalize = normalizeAdaptor(Button);
    const { props } = normalize.editor('normal');
    expect(getDefaultProps(normalize, 'normal')).toEqual({shape: 'normal', 'level': 'normal', 'size': 'medium', 'block': false, data: 'Hello'});

    expect(findProp('size', props).name).toBe('size');
    expect(findOptionLabel('medium', 'size', props)).toBe('Medium');
  });

  it('should be generateDemos', () => {
    expect(generateDemos(normalizeAdaptor(Button)).map(item => {
      delete item.id;
      return item;
    })).toMatchSnapshot();
  })
});
