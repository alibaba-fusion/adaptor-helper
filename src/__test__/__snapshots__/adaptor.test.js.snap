// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`adaptor.js should be generateDemos 1`] = `
Array [
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "block": false,
        "data": "Hello",
        "level": "normal",
        "shape": "normal",
        "size": "large",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "block": false,
        "data": "Hello",
        "level": "primary",
        "shape": "normal",
        "size": "large",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "block": false,
        "data": "Hello",
        "level": "secondary",
        "shape": "normal",
        "size": "large",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "block": false,
        "data": "Hello",
        "level": "warning",
        "shape": "normal",
        "size": "large",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "block": false,
        "data": "Hello",
        "level": "light",
        "shape": "normal",
        "size": "large",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "block": false,
        "data": "Hello",
        "level": "dark",
        "shape": "normal",
        "size": "large",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "block": false,
        "data": "Hello",
        "level": "normal",
        "shape": "normal",
        "size": "medium",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "block": false,
        "data": "Hello",
        "level": "primary",
        "shape": "normal",
        "size": "medium",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "block": false,
        "data": "Hello",
        "level": "secondary",
        "shape": "normal",
        "size": "medium",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "block": false,
        "data": "Hello",
        "level": "warning",
        "shape": "normal",
        "size": "medium",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "block": false,
        "data": "Hello",
        "level": "light",
        "shape": "normal",
        "size": "medium",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "block": false,
        "data": "Hello",
        "level": "dark",
        "shape": "normal",
        "size": "medium",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "block": false,
        "data": "Hello",
        "level": "normal",
        "shape": "normal",
        "size": "small",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "block": false,
        "data": "Hello",
        "level": "primary",
        "shape": "normal",
        "size": "small",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "block": false,
        "data": "Hello",
        "level": "secondary",
        "shape": "normal",
        "size": "small",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "block": false,
        "data": "Hello",
        "level": "warning",
        "shape": "normal",
        "size": "small",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "block": false,
        "data": "Hello",
        "level": "light",
        "shape": "normal",
        "size": "small",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "block": false,
        "data": "Hello",
        "level": "dark",
        "shape": "normal",
        "size": "small",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "block": false,
        "data": "~Hello",
        "level": "normal",
        "shape": "normal",
        "size": "large",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "block": false,
        "data": "~Hello",
        "level": "primary",
        "shape": "normal",
        "size": "large",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "block": false,
        "data": "~Hello",
        "level": "secondary",
        "shape": "normal",
        "size": "large",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "block": false,
        "data": "~Hello",
        "level": "warning",
        "shape": "normal",
        "size": "large",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "block": false,
        "data": "~Hello",
        "level": "light",
        "shape": "normal",
        "size": "large",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "block": false,
        "data": "~Hello",
        "level": "dark",
        "shape": "normal",
        "size": "large",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "block": false,
        "data": "~Hello",
        "level": "normal",
        "shape": "normal",
        "size": "medium",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "block": false,
        "data": "~Hello",
        "level": "primary",
        "shape": "normal",
        "size": "medium",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "block": false,
        "data": "~Hello",
        "level": "secondary",
        "shape": "normal",
        "size": "medium",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "block": false,
        "data": "~Hello",
        "level": "warning",
        "shape": "normal",
        "size": "medium",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "block": false,
        "data": "~Hello",
        "level": "light",
        "shape": "normal",
        "size": "medium",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "block": false,
        "data": "~Hello",
        "level": "dark",
        "shape": "normal",
        "size": "medium",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "block": false,
        "data": "~Hello",
        "level": "normal",
        "shape": "normal",
        "size": "small",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "block": false,
        "data": "~Hello",
        "level": "primary",
        "shape": "normal",
        "size": "small",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "block": false,
        "data": "~Hello",
        "level": "secondary",
        "shape": "normal",
        "size": "small",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "block": false,
        "data": "~Hello",
        "level": "warning",
        "shape": "normal",
        "size": "small",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "block": false,
        "data": "~Hello",
        "level": "light",
        "shape": "normal",
        "size": "small",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "block": false,
        "data": "~Hello",
        "level": "dark",
        "shape": "normal",
        "size": "small",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "block": false,
        "data": "*Hello",
        "level": "normal",
        "shape": "normal",
        "size": "large",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "block": false,
        "data": "*Hello",
        "level": "primary",
        "shape": "normal",
        "size": "large",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "block": false,
        "data": "*Hello",
        "level": "secondary",
        "shape": "normal",
        "size": "large",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "block": false,
        "data": "*Hello",
        "level": "warning",
        "shape": "normal",
        "size": "large",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "block": false,
        "data": "*Hello",
        "level": "light",
        "shape": "normal",
        "size": "large",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "block": false,
        "data": "*Hello",
        "level": "dark",
        "shape": "normal",
        "size": "large",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "block": false,
        "data": "*Hello",
        "level": "normal",
        "shape": "normal",
        "size": "medium",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "block": false,
        "data": "*Hello",
        "level": "primary",
        "shape": "normal",
        "size": "medium",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "block": false,
        "data": "*Hello",
        "level": "secondary",
        "shape": "normal",
        "size": "medium",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "block": false,
        "data": "*Hello",
        "level": "warning",
        "shape": "normal",
        "size": "medium",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "block": false,
        "data": "*Hello",
        "level": "light",
        "shape": "normal",
        "size": "medium",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "block": false,
        "data": "*Hello",
        "level": "dark",
        "shape": "normal",
        "size": "medium",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "block": false,
        "data": "*Hello",
        "level": "normal",
        "shape": "normal",
        "size": "small",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "block": false,
        "data": "*Hello",
        "level": "primary",
        "shape": "normal",
        "size": "small",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "block": false,
        "data": "*Hello",
        "level": "secondary",
        "shape": "normal",
        "size": "small",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "block": false,
        "data": "*Hello",
        "level": "warning",
        "shape": "normal",
        "size": "small",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "block": false,
        "data": "*Hello",
        "level": "light",
        "shape": "normal",
        "size": "small",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "block": false,
        "data": "*Hello",
        "level": "dark",
        "shape": "normal",
        "size": "small",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "block": false,
        "data": "-Hello",
        "level": "normal",
        "shape": "normal",
        "size": "large",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "block": false,
        "data": "-Hello",
        "level": "primary",
        "shape": "normal",
        "size": "large",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "block": false,
        "data": "-Hello",
        "level": "secondary",
        "shape": "normal",
        "size": "large",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "block": false,
        "data": "-Hello",
        "level": "warning",
        "shape": "normal",
        "size": "large",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "block": false,
        "data": "-Hello",
        "level": "light",
        "shape": "normal",
        "size": "large",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "block": false,
        "data": "-Hello",
        "level": "dark",
        "shape": "normal",
        "size": "large",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "block": false,
        "data": "-Hello",
        "level": "normal",
        "shape": "normal",
        "size": "medium",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "block": false,
        "data": "-Hello",
        "level": "primary",
        "shape": "normal",
        "size": "medium",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "block": false,
        "data": "-Hello",
        "level": "secondary",
        "shape": "normal",
        "size": "medium",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "block": false,
        "data": "-Hello",
        "level": "warning",
        "shape": "normal",
        "size": "medium",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "block": false,
        "data": "-Hello",
        "level": "light",
        "shape": "normal",
        "size": "medium",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "block": false,
        "data": "-Hello",
        "level": "dark",
        "shape": "normal",
        "size": "medium",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "block": false,
        "data": "-Hello",
        "level": "normal",
        "shape": "normal",
        "size": "small",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "block": false,
        "data": "-Hello",
        "level": "primary",
        "shape": "normal",
        "size": "small",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "block": false,
        "data": "-Hello",
        "level": "secondary",
        "shape": "normal",
        "size": "small",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "block": false,
        "data": "-Hello",
        "level": "warning",
        "shape": "normal",
        "size": "small",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "block": false,
        "data": "-Hello",
        "level": "light",
        "shape": "normal",
        "size": "small",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "block": false,
        "data": "-Hello",
        "level": "dark",
        "shape": "normal",
        "size": "small",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "data": "Hello",
        "level": "normal",
        "shape": "text",
        "size": "medium",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "data": "Hello",
        "level": "primary",
        "shape": "text",
        "size": "medium",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "data": "Hello",
        "level": "normal",
        "shape": "text",
        "size": "small",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "data": "Hello",
        "level": "primary",
        "shape": "text",
        "size": "small",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "data": "~Hello",
        "level": "normal",
        "shape": "text",
        "size": "medium",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "data": "~Hello",
        "level": "primary",
        "shape": "text",
        "size": "medium",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "data": "~Hello",
        "level": "normal",
        "shape": "text",
        "size": "small",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "data": "~Hello",
        "level": "primary",
        "shape": "text",
        "size": "small",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "data": "*Hello",
        "level": "normal",
        "shape": "text",
        "size": "medium",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "data": "*Hello",
        "level": "primary",
        "shape": "text",
        "size": "medium",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "data": "*Hello",
        "level": "normal",
        "shape": "text",
        "size": "small",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "data": "*Hello",
        "level": "primary",
        "shape": "text",
        "size": "small",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "data": "-Hello",
        "level": "normal",
        "shape": "text",
        "size": "medium",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "data": "-Hello",
        "level": "primary",
        "shape": "text",
        "size": "medium",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "data": "-Hello",
        "level": "normal",
        "shape": "text",
        "size": "small",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "data": "-Hello",
        "level": "primary",
        "shape": "text",
        "size": "small",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "data": "Hello",
        "level": "normal",
        "shape": "list",
        "size": "medium",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "data": "Hello",
        "level": "primary",
        "shape": "list",
        "size": "medium",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "data": "Hello",
        "level": "secondary",
        "shape": "list",
        "size": "medium",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "data": "~Hello",
        "level": "normal",
        "shape": "list",
        "size": "medium",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "data": "~Hello",
        "level": "primary",
        "shape": "list",
        "size": "medium",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "data": "~Hello",
        "level": "secondary",
        "shape": "list",
        "size": "medium",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "data": "*Hello",
        "level": "normal",
        "shape": "list",
        "size": "medium",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "data": "*Hello",
        "level": "primary",
        "shape": "list",
        "size": "medium",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "data": "*Hello",
        "level": "secondary",
        "shape": "list",
        "size": "medium",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "data": "-Hello",
        "level": "normal",
        "shape": "list",
        "size": "medium",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "data": "-Hello",
        "level": "primary",
        "shape": "list",
        "size": "medium",
      },
    },
  },
  Object {
    "background": "transparent",
    "node": Object {
      "adaptor": "Button",
      "props": Object {
        "data": "-Hello",
        "level": "secondary",
        "shape": "list",
        "size": "medium",
      },
    },
  },
]
`;

exports[`adaptor.js should be normalizeAdaptor 1`] = `
Object {
  "adaptor": [Function],
  "content": [Function],
  "demoOptions": [Function],
  "editor": [Function],
  "name": "Button",
  "shape": Array [
    Object {
      "label": "Normal ",
      "value": "normal",
    },
    Object {
      "label": "Text Button",
      "value": "text",
    },
    Object {
      "label": "List Button",
      "value": "list",
    },
  ],
}
`;

exports[`adaptor.js should be normalizeAdaptor 2`] = `
Object {
  "data": Object {
    "active": true,
    "default": "Hello",
    "disable": true,
    "hover": true,
    "icon": true,
  },
  "nodes": Array [],
  "props": Array [
    Object {
      "default": "normal",
      "label": "Level",
      "name": "level",
      "options": Array [
        Object {
          "label": "Normal",
          "value": "normal",
        },
        Object {
          "label": "Primary",
          "value": "primary",
        },
        Object {
          "label": "Secondary",
          "value": "secondary",
        },
        Object {
          "label": "Warning",
          "value": "warning",
        },
        Object {
          "label": "Light",
          "value": "light",
        },
        Object {
          "label": "Dark",
          "value": "dark",
        },
      ],
      "type": "enum",
    },
    Object {
      "default": "medium",
      "label": "Size",
      "name": "size",
      "options": Array [
        Object {
          "label": "Large",
          "value": "large",
        },
        Object {
          "label": "Medium",
          "value": "medium",
        },
        Object {
          "label": "Small",
          "value": "small",
        },
      ],
      "type": "enum",
    },
    Object {
      "label": "Block",
      "name": "block",
      "options": Array [],
      "type": "bool",
    },
  ],
}
`;

exports[`adaptor.js should be normalizeAdaptor 3`] = `
Object {
  "options": Array [
    Object {
      "default": "none",
      "label": "Icon",
      "name": "icon",
      "options": Array [
        Object {
          "label": "None",
          "value": "none",
        },
        Object {
          "label": "Left",
          "value": "left",
        },
        Object {
          "label": "Right",
          "value": "right",
        },
      ],
    },
    Object {
      "default": "no",
      "label": "Group",
      "name": "group",
      "options": Array [
        Object {
          "label": "Yes",
          "value": "yes",
        },
        Object {
          "label": "No",
          "value": "no",
        },
      ],
    },
  ],
  "transform": [Function],
}
`;

exports[`adaptor.js should be normalizeAdaptor 4`] = `
Object {
  "data": Object {
    "active": true,
    "default": "Hello",
    "disable": true,
    "hover": true,
    "icon": true,
  },
  "nodes": Array [],
  "props": Array [
    Object {
      "default": "normal",
      "label": "Level",
      "name": "level",
      "options": Array [
        Object {
          "label": "Normal",
          "value": "normal",
        },
        Object {
          "label": "Primary",
          "value": "primary",
        },
      ],
      "type": "enum",
    },
    Object {
      "default": "medium",
      "label": "Size",
      "name": "size",
      "options": Array [
        Object {
          "label": "Medium",
          "value": "medium",
        },
        Object {
          "label": "Small",
          "value": "small",
        },
      ],
      "type": "enum",
    },
  ],
}
`;

exports[`adaptor.js should be normalizeAdaptor 5`] = `
Object {
  "options": Array [
    Object {
      "default": "none",
      "label": "Icon",
      "name": "icon",
      "options": Array [
        Object {
          "label": "None",
          "value": "none",
        },
        Object {
          "label": "Left",
          "value": "left",
        },
        Object {
          "label": "Right",
          "value": "right",
        },
      ],
    },
  ],
  "transform": [Function],
}
`;

exports[`adaptor.js should be normalizeAdaptor 6`] = `
Object {
  "data": Object {
    "active": true,
    "default": "Hello",
    "disable": true,
    "hover": true,
    "icon": true,
  },
  "nodes": Array [],
  "props": Array [
    Object {
      "default": "normal",
      "label": "Level",
      "name": "level",
      "options": Array [
        Object {
          "label": "Normal",
          "value": "normal",
        },
        Object {
          "label": "Primary",
          "value": "primary",
        },
        Object {
          "label": "Secondary",
          "value": "secondary",
        },
      ],
      "type": "enum",
    },
    Object {
      "default": "medium",
      "label": "Size",
      "name": "size",
      "options": Array [
        Object {
          "label": "Medium",
          "value": "medium",
        },
      ],
      "type": "enum",
    },
  ],
}
`;

exports[`adaptor.js should be normalizeAdaptor 7`] = `
Object {
  "options": Array [
    Object {
      "default": "none",
      "label": "Icon",
      "name": "icon",
      "options": Array [
        Object {
          "label": "None",
          "value": "none",
        },
        Object {
          "label": "Left",
          "value": "left",
        },
        Object {
          "label": "Right",
          "value": "right",
        },
      ],
    },
    Object {
      "default": "no",
      "label": "Group",
      "name": "group",
      "options": Array [
        Object {
          "label": "Yes",
          "value": "yes",
        },
        Object {
          "label": "No",
          "value": "no",
        },
      ],
    },
  ],
  "transform": [Function],
}
`;
